### PR TITLE
Add button to append vocababulary markdown syntax

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds a button to the ActivitySectionCard to append the vocabulary syntax (`[v <key>]`).

https://user-images.githubusercontent.com/46464143/105220813-17fb9c00-5b0d-11eb-91bb-7b91067ee784.mov

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCard.jsx [hunk 1]
```diff
@@ -352,16 +352,24 @@ class ActivitySectionCard extends Component {
...
 
+  appendResourceLink = resourceKey => {
+    this.appendMarkdownSyntax(`[r ${resourceKey}]`);
+  };
+
+  appendVocabularyLink = vocabularyKey => {
+    this.appendMarkdownSyntax(`[v ${vocabularyKey}]`);
+  };
+
   appendSlide = () => {
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCard.jsx:374]: Should appendSlide use the new appendMarkdown function?

File: apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCard.jsx [hunk 2]
```diff
@@ -352,16 +352,24 @@ class ActivitySectionCard extends Component {
     );
   };
 
-  appendResourceLink = resourceKey => {
+  appendMarkdownSyntax = strToAppend => {
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCard.jsx:356]: Nice extraction!

File: apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx [hunk 3]
```diff
@@ -161,6 +175,16 @@ class ActivitySectionCardButtons extends Component {
...
               Resource Link
             </button>
+            <button
+              onMouseDown={this.handleOpenAddVocabulary}
+              className="btn"
+              style={styles.addButton}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Vocabulary Link
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx]: I think to save space we should have this button just say "Vocab" and probably want to change Resource Link to Resource. There are just too many buttons...

File: apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    handleConfirm: PropTypes.func.isRequired,
+    handleClose: PropTypes.func.isRequired,
+    vocabularies: PropTypes.arrayOf(vocabularyShape)
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedVocabularyKey:
+        this.props.vocabularies.length > 0 ? this.props.vocabularies[0].key : ''
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx:29]: Is there are no vocabularies should we just disable the button instead?

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx:29]: Probably also should consider this for resources but that doesn't have to happen here.

**Review by dmcavoy — APPROVED** [review]
Looks good! I left a couple suggestions of things i think might improve the experience which would be good to consider before merging.

File: apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx (hunk 4 above)
Comment (davidsbailey) [reply apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx:29]: I like this -- some courses may not even use Vocab (or Resources), so this sounds like a great way to reduce clutter.

**Review by davidsbailey — APPROVED** [review]
LGTM after addressing Dani's comments

File: apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx (hunk 4 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx:29]: I like that! I'll do that, at least for vocabulary in this PR.

File: apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx (hunk 3 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx]: Makes sense. I guess the only reason I could see to keep "Vocabulary Link" is that this button adds a link, not a vocab, but I think that'll be pretty clear.

File: apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx (hunk 4 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/lesson-editor/FindVocabularyDialog.jsx:29]: Done for vocab -- I'll do this for resources in a follow-up PR

**Pull request merged**